File: app/src/main/java/com/example/wagtailjournal/Musume.java
package com.example.wagtailjournal;

import java.io.File;

public class Musume {
    public File searchLatestJournal(File directory) {
        File found = null;
        if (!directory.exists())
            return null;
        for (File file : directory.listFiles()) {
            if (found == null) {
                found = file;
                continue;
            }
            if (file.lastModified() > found.lastModified()) {
                found = file;
            }
        }
        return found;
    }
}

File: app/src/main/java/com/github/tkojitu/wagtailjournal/Musume.kt
package com.github.tkojitu.wagtailjournal

import java.io.File

class Musume {
    fun searchLatestJournal(directory: File): File? {
        var found: File? = null
        if (!directory.exists()) return null
        for (file in directory.listFiles()!!) {
            if (found == null) {
                found = file
                continue
            }
            if (file.lastModified() > found.lastModified()) {
                found = file
            }
        }
        return found
    }
}
